framework_tvb/tvb/adapters/visualizers/surface_view.py

```diff
@@ -386,14 +387,15 @@ def _compute_measure_param(connectivity_measure, measure_points_no):
             max_measure = measure_points_no
             client_measure_url = ''
         else:
-            connectivity_measure_shape = connectivity_measure.array_data.shape
+            conn_array = connectivity_measure.array_data.load()
+            connectivity_measure_shape = conn_array.shape
```

Comment:
Can't we read the shape from ConnectivityMeasureIndex?
Also min&max values should be found in the metadata.

Reply:
Currently, we don't keep the shape on ConnectivityMeasureIndex, but for this use-case it might help to keep it. We do keep min&max.

Reply:
I see now, that `ConnectivityMeasureIndex` has only min/max/mean.
`DataTypeMatrix` has only `ndim`, but I recommend adding the shape also there.